### DISCUSSION TITLE
Where to find FDS source code of velocity, temperature and species transport equation

### DISCUSSION THREAD

**Discussion by AlanZandie** [firemodels/fds#12572] [General]
Good day everyone,

I am Zandie, from UoB, UK, I have been meaning to apply certain optimisation techniques on FDS codes to minimise/maximise a few features I have been dealing with. E.g. I am currently simulating a car park where I want to find out the best fans configurations under which the CO concentration, fan velocities, and temperature, of the premise take the optimum values that I am looking for. However, in doing so, I need to know where is the script line to in FDS where those features are being calculated, so I could recall them in my code. 
Could you please direct me to a particular file in the source package to access the relevant scripts?

Thank you,

**Comment by drjfloyd** [comment]
When you say "script line" 

-Are you referring to an input that you put in the .fds input file? In that case please read the User's Guide which describes the various inputs and outputs for FDS.

-Are you referring to the a line of Fortran code in the FDS Source.  In that case there isn't a single file or a single line of source code that computes each of temperature, velocity, and species in FDS. That is what the 100,000+ lines of source code does as a whole.  The major files for transport are pres.f90, velof.f90, divg.f90, and mass.f90 but these may also rely upon the heat release rate (fire.f90), radiation heat transfer (radi.f90), wall heat transfer (wall.f90), HVAC flows (hvac.f90), and sprinkler sprays (part.f90).

**Reply by AlanZandie** [reply]
I understand the complexity of the written FDS codes, I was pretty much looking for the files you just mentioned. 
Thank you for your help.
